plugins/russian/LitSpace.ts

```diff
@@ -1,241 +1,242 @@
-    },
-  } satisfies Filters;
-}
-
-export default new freedlit();
+import { Plugin } from "@typings/plugin";
+import { FilterTypes, Filters } from "@libs/filterInputs";
+import { fetchApi, fetchFile } from "@libs/fetch";
+import { NovelStatus } from "@libs/novelStatus";
+import { load as parseHTML } from "cheerio";
+
+class freedlit implements Plugin.PluginBase {
+  id = "freedlit.space";
+  name = "LitSpace";
+  site = "https://freedlit.space";
+  version = "1.0.0";
+  icon = "src/ru/freedlit/icon.png";
+
+  async popularNovels(
+    page: number,
+    { showLatestNovels, filters }: Plugin.PopularNovelsOptions<typeof this.filters>,
+  ): Promise<Plugin.NovelItem[]> {
+    let url = this.site + "/get-books/all/list/" + page + "?sort=";
+    url += showLatestNovels ? "recent" : filters?.sort?.value || "popular";
+    url += "&status=" + (filters?.status?.value || "all");
+    url += "&access=" + (filters?.access?.value || "all");
+    url += "&adult=" + (filters?.adult?.value || "hide");
+
+    if (filters?.genre?.value?.include?.length) {
+      url += filters.genre.value.include
+        .map((id) => "&genres_included[]=" + id)
+        .join("");
+    }
+
+    if (filters?.genre?.value?.exclude?.length) {
+      url += filters.genre.value.exclude
+        .map((id) => "&genres_excluded[]=" + id)
+        .join("");
+    }
+
+    const body = await fetchApi(url).then((res) => res.text());
+    const loadedCheerio = parseHTML(body);
+
+    const novels: Plugin.NovelItem[] = [];
+    loadedCheerio("#bookListBlock > div > div").each(function () {
+      const name = loadedCheerio(this).find("div > h4 > a").text()?.trim();
+      const cover = loadedCheerio(this)
+        .find("div > a > img")
+        .attr("src")
+        ?.trim();
+      const url = loadedCheerio(this).find("div > h4 > a").attr("href")?.trim();
+      if (!name || !url) return;
+
+      novels.push({ name, cover, url });
+    });
+
+    return novels;
+  }
+
+  async parseNovelAndChapters(novelUrl: string): Promise<Plugin.SourceNovel> {
+    const body = await fetchApi(novelUrl).then((res) => res.text());
+    const loadedCheerio = parseHTML(body);
+
+    const novel: Plugin.SourceNovel = {
+      url: novelUrl,
+      name: loadedCheerio(".book-info h4").text(),
+      cover: loadedCheerio(".book-cover > div > img").attr("src")?.trim(),
+      summary: loadedCheerio("#nav-home").text()?.trim(),
+      author: loadedCheerio(".book-info h5 > a").text(),
+      genres: loadedCheerio(".genre-list > a")
+        .map((index, element) => loadedCheerio(element).text())
+        .get()
+        .join(","),
+    };
+
+    const chapters: Plugin.ChapterItem[] = [];
+
+    loadedCheerio("a.chapter-line").each((chapterIndex, element) => {
+      const name = loadedCheerio(element).find("h6").text();
+      const url = loadedCheerio(element).attr("href");
+      if (!name || !url) return;
+
+      const releaseTime = loadedCheerio(element).find('span[class="date"]').text();
```

Comment:
could you parse this into ISO string format.

Reply:
Perhaps it would be better to pass an instance of Date?

Reply:
it's not mandatory, we just want a consistent Date system xD